types/block.go

```diff
@@ -309,6 +309,7 @@ func NewBlock(header *Header, txs []*Transaction, lastCommit *Commit, evidence [
 		b.header.EvidenceHash = common.NewZeroHash()
 	}
 
+	b.Hash()
```

Comment:
Add comment and define its purpose

Reply:
I removed it. Fixed in blockchain reactor test suite.